tests/app/views/test_drafts.py

```diff
@@ -887,3 +910,93 @@ def test_should_not_create_draft_with_invalid_data(self):
+                    supplier_id=1,
+                    contact_name=u"Test",
+                    email=u"supplier@user.dmdev",
+                    postcode=u"SW1A 1AA"
+                )
+            )
+            Framework.query.filter_by(slug='g-cloud-7').update(dict(status='open'))
+            db.session.commit()
+
+        create_draft_json = {
+            'update_details': {
+                'updated_by': 'joeblogs'
+            },
+            'services': self.load_example_listing("G7-SCS")
+        }
+
+        draft = self.client.post(
+            '/draft-services/g-cloud-7/create',
+            data=json.dumps(create_draft_json),
+            content_type='application/json')
+
+        self.draft = json.loads(draft.get_data())['services']
+        self.draft_id = self.draft['id']
+
+    def test_complete_draft(self):
+        res = self.client.post(
+            '/draft-services/%s/complete' % self.draft_id,
+            data=json.dumps({'update_details': {'updated_by': 'joeblogs'}}),
+            content_type='application/json')
+
+        data = json.loads(res.get_data())
+        assert_equal(res.status_code, 200)
+        assert_equal(data['services']['status'], 'submitted')
+
+    def test_complete_draft_should_create_audit_event(self):
+        res = self.client.post(
+            '/draft-services/%s/complete' % self.draft_id,
+            data=json.dumps({'update_details': {'updated_by': 'joeblogs'}}),
+            content_type='application/json')
+
+        assert_equal(res.status_code, 200)
+
+        audit_response = self.client.get('/audit-events')
+        assert_equal(audit_response.status_code, 200)
+        data = json.loads(audit_response.get_data())
+        assert_equal(len(data['auditEvents']), 2)
+        assert_equal(data['auditEvents'][1]['user'], 'joeblogs')
+        assert_equal(data['auditEvents'][1]['type'], 'complete_draft_service')
+        assert_equal(data['auditEvents'][1]['data'], {
+            'draftId': self.draft_id,
+        })
+
+    def test_should_not_complete_invalid_draft(self):
+        create_draft_json = {
+            'update_details': {
+                'updated_by': 'joeblogs'
+            },
+            'services': {
+                'lot': 'SCS',
+                'supplierId': 1,
+                'serviceName': 'Name',
+            }
+        }
+
+        draft = self.client.post(
+            '/draft-services/g-cloud-7/create',
+            data=json.dumps(create_draft_json),
+            content_type='application/json'
+        )
+
+        draft = json.loads(draft.get_data())['services']
+
+        res = self.client.post(
+            '/draft-services/%s/complete' % draft['id'],
+            data=json.dumps({'update_details': {'updated_by': 'joeblogs'}}),
+            content_type='application/json')
+
+        assert_equal(res.status_code, 400)
+        errors = json.loads(res.get_data())['error']
+        assert_in('serviceSummary', errors)
```

Comment:
Should we have a test that confirms it fails if no updater details are provided?